### PR TITLE
web: Add registerMerchant to merchant creation

### PR DESCRIPTION
**Pull request by backspace**
This includes placeholders for merchant customisation and the actual
choosing of the prepaid card.

### CODE REVIEW

File: packages/web-client/app/components/card-pay/create-merchant-workflow/prepaid-card-choice/index.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  TransactionHash,
+} from '@cardstack/web-client/utils/web3-strategies/types';
+
+import { task, TaskGenerator } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+import { reads } from 'macro-decorators';
+
+// http://ember-concurrency.com/docs/typescript
+// infer whether we should treat the return of a yield statement as a promise
+type Resolved<T> = T extends PromiseLike<infer R> ? R : T;
```
Comment (lukemelia) [inline packages/web-client/app/components/card-pay/create-merchant-workflow/prepaid-card-choice/index.ts:23]: This is also defined in app/components/card-pay/issue-prepaid-card-workflow/preview/index.ts. Let's extract it and define it in one place.

File: packages/web-client/app/components/card-pay/create-merchant-workflow/prepaid-card-choice/index.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      .perform()
+      .catch((e) => console.error(e));
+  }
+
+  @task *createTask(): TaskGenerator<void> {
+    let { workflowSession } = this.args;
+
+    try {
+      this.chinInProgressMessage =
+        'Preparing to create your custom prepaid card…';
```
Comment (lukemelia) [inline packages/web-client/app/components/card-pay/create-merchant-workflow/prepaid-card-choice/index.ts:50]: This message is not correct.

**Review by lukemelia — CHANGES_REQUESTED** [review]

File: packages/web-client/app/components/card-pay/create-merchant-workflow/prepaid-card-choice/index.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+          this.txHash = txHash;
+          this.chinInProgressMessage =
+            'Waiting for the transaction to be finalized…';
+        },
+      };
+
+      let prepaidCards = (yield taskFor(
+        this.layer2Network.refreshSafes
+      ).perform()).filterBy('type', 'prepaid-card');
+      let placeholderPrepaidCard = prepaidCards[0]!;
```
Comment (Aierie) [inline packages/web-client/app/components/card-pay/create-merchant-workflow/prepaid-card-choice/index.ts:69]: Given that this card should allow the user to select a prepaid card, would expect the fetching to occur earlier in the lifeycle of the card and have the fetched prepaid cards be stored as a property of this component. Probably can be done when there is UI follow-up!

**Review by Aierie — APPROVED** [review]
I think the refactors + copy change that Luke has suggested would be good! Otherwise, this seems like a good setup for the prepaid card selection UI that will follow.

File: packages/web-client/app/components/card-pay/create-merchant-workflow/prepaid-card-choice/index.ts (hunk 1 above)
Comment (backspace) [reply packages/web-client/app/components/card-pay/create-merchant-workflow/prepaid-card-choice/index.ts]: 👍🏻 I extracted it to `utils/resolved` 😀

File: packages/web-client/app/components/card-pay/create-merchant-workflow/prepaid-card-choice/index.ts (hunk 2 above)
Comment (backspace) [reply packages/web-client/app/components/card-pay/create-merchant-workflow/prepaid-card-choice/index.ts]: hah indeed, updated now to `Preparing to create merchant…`

File: packages/web-client/app/components/card-pay/create-merchant-workflow/prepaid-card-choice/index.ts (hunk 3 above)
Comment (backspace) [reply packages/web-client/app/components/card-pay/create-merchant-workflow/prepaid-card-choice/index.ts:66]: ya this is just choosing a placeholder to be passed in to the SDK call, fetching can indeed happen earlier to list the available cards

**Pull request merged**